R/preprocessing.R

```diff
@@ -645,6 +645,133 @@ Load10X_Spatial <- function (
 
   return(object)
 }
+#' Add 10X Cell Types to a Seurat Object
+#'
+#' This function reads cell type annotations from a CSV file and adds them to the metadata of a Seurat object.
+#' If the cell type file does not exist, the original Seurat object is returned unchanged.
+#'
+#' @param data.dir A string specifying the directory containing the "cell_types" folder with the "cell_types.csv" file.
+#' @param object A Seurat object to which the cell type annotations will be added.
+#'
+#' @return A Seurat object with updated metadata including cell type annotations if the file is found.
+#'
+#' @details
+#' The function searches for a CSV file named "cell_types.csv" in the "cell_types" subdirectory within `data.dir`.
+#' The CSV file should contain at least a "barcode" column that matches the cell barcodes in the Seurat object.
+#' Additional columns in the CSV file will be merged into the Seurat object's metadata.
+#'
+#' @importFrom utils read.csv
+#' @importFrom tibble rownames_to_column column_to_rownames
+#' @importFrom base file.path file.exists merge
+#'
+#' @examples
+#' \dontrun{
+#' # Specify the data directory containing the "cell_types" folder
+#' data.dir <- "/path/to/data"
+#'
+#' # Create a Seurat object (example)
+#' seurat_obj <- CreateSeuratObject(counts = some_counts_matrix)
+#'
+#' # Add cell type annotations to the Seurat object
+#' seurat_obj <- Add_10X_CellTypes(data.dir, seurat_obj)
+#' }
+
+Add_10X_CellTypes <- function(data.dir, object) {
+  cell_types_path <- file.path(data.dir, "cell_types", "cell_types.csv")
+  if (file.exists(cell_types_path)) {
+    cell.types <- read.csv(cell_types_path)
+    object@meta.data <- dplyr::left_join(tibble::rownames_to_column(object@meta.data, "barcode"),
+                                         cell.types, by = "barcode") %>%
+      tibble::column_to_rownames("barcode")
+    return(object)
+  } else {
+    return(object)
+  }
+}
+
+#' Load a 10x Genomics Single Cell Experiment into a \code{Seurat} object
+#'
+#' @inheritParams Read10X
+#' @inheritParams SeuratObject::CreateSeuratObject
```

Comment:
Might be worth noting that the supplied `assay` argument won't be used if the h5 is multimodal and returns a list.

Reply:
will make a note about the assay. ultimately we don't return a list but a single seurat object so I don't want to add so much detail as to be confusing.